11-if/02-if-statement/05-challenge-userpass/02-2nd-challenge/01-challenge/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	usage     = "Usage: [username] [password]"
+	errUser   = "Access denied for %q.\n"
+	errPwd    = "Invalid password for %q.\n"
+	accessOK  = "Access granted to %q.\n"
+	user      = "jack"
+	pass      = "1888"
+	user2     = "inanc"
+	password2 = "1879"
+)
+
+func main() {
+	args := os.Args
+
+	if len(args) != 3 {
+		fmt.Println(usage)
+		return
+	}
+
+	u, p := args[1], args[2]
+
+	if u != user && u != user2 {
```

Comment:
let's improve this example

move the logic for checking passwords to its own function like this

```go
type CheckAssessResult int

const (
   Allowed CheckAssessResult = 1 + iota
   InvalidPassword
   AccessDenied
)

func checkAccess(username, password string, desiredUsername, desiredPassword string) CheckAssessResult {
  // check logic here
}
```
also implement a function to show results
```go
func printResult(result CheckAssessResult, username string) {
// print logic here
}
```

and use it smth like this
NOTE: this 2 IF conditions are not optimal and could be replaced with a single check - think how to improve it
```go
result := checkAccess(u, p, user, pass)
if result == Allowed {
  printResult(result, u)
  return 
}

if result == InvalidPassword {
  printResult(result, u)
  return 
}



result = checkAccess(u, p, user2, pass2)
printResult(result, u)

```